src/Microsoft.ML.ImageAnalytics/ImagePixelExtractor.cs

```diff
@@ -416,43 +417,40 @@ private ValueGetter<VBuffer<TValue>> GetGetterCore<TValue>(DataViewRow input, in
                             int idstMin = 0;
                             for (int y = 0; y < h; ++y)
                             {
-                                int idstBase = idstMin + y * w;
+                                int idst = idstMin + y * w;
 
-                                // Note that the bytes are in order BGR[A]. We arrange the layers in order ARGB.
                                 if (!vb.IsEmpty)
                                 {
-                                    for (int x = 0; x < w; x++, idstBase++)
+                                    for (int x = 0; x < w; x++, idst++)
                                     {
```

Comment:
If you refactor this by having the `if...else if...else` loop inside the for loop you can simplify the code a bit. It would look like:
`for... {
     var pb = src.GetPixel(x,y);
     if...
     else if...
     else...
}`